python/lsst/ts/observatory/control/maintel/mtcalsys.py

```diff
@@ -0,0 +1,567 @@
+        filter_name = config_data[
+            "filter"
+        ]  # not sure if this is an enumeration somewhere?
+
+        task_setup_camera = (
+            self.mtcamera.setup_instrument(
+                filter=config_data["atspec_filter"],
+                grating=config_data["atspec_grating"],
+            )
+            if self.mtcamera is not None and config_data["use_camera"]
+            else utils.make_done_future()
+        )
+        if self.mtcamera is None and config_data["use_camera"]:
+            raise RuntimeError(
+                f"MTCamera is not defined but {config_name} requires it. "
+                "Make sure you are instantiating ComCam and passing it to MTCalsys."
+            )
+
+        if calibration_type == CalibrationType.WhiteLight:
+            leds_ = self.led_per_filter[filter_name]
+            task_select_wavelength = (
+                self.linearstage_led_select.cmd_moveAbsolute.set_start(
+                    distance=self.ls_led_locations[filter_name]
+                )
+            )
+            task_adjust_focus = self.linearstage_led_focus.cmd_moveAbsolute.set_start(
+                distance=self.ls_led_focus[filter_name]
+            )
+            task_turn_on = self.rem.ledprojector.cmd_switchOn.set_start(
+                serialNumbers=leds_
+            )
+
+            await asyncio.gather(
+                task_select_wavelength,
+                task_adjust_focus,
+                task_turn_on,
+                task_setup_camera,
+            )
+
+        elif calibration_type == CalibrationType.Mono:
+            wavelengths = [400.0]  # function of filter_name
+            task_select_wavelength = self.change_wavelength(wavelength=wavelengths[0])
+
+            await asyncio.gather(task_select_wavelength, task_setup_camera)
+
+    async def run_calibration_sequence(
+        self, sequence_name: str, exposure_metadata: dict
+    ) -> dict:
+        """Perform full calibration sequence, taking flats with the
+        camera and all ancillary instruments.
+
+        Parameters
+        ----------
+        sequence_name : `str`
+            Name of the calibration sequence to execute.
+
+        Returns
+        -------
+        calibration_summary : `dict`
+            Dictionary with summary information about the sequence.
+        """
+        calibration_summary: dict = dict(
+            steps=[],
+            sequence_name=sequence_name,
+        )
+
+        config_data = self.get_calibration_configuration(sequence_name)
+
+        calibration_type = getattr(CalibrationType, str(config_data["calib_type"]))
+        if calibration_type == CalibrationType.WhiteLight:
+            calibration_wavelengths = np.array([float(config_data["wavelength"])])
+        else:
+            wavelength = float(config_data["wavelength"])
+            wavelength_width = float(
+                config_data["wavelength_width"]
+            )  # This doesn't make any sense
```

Comment:
What doesn't make any sense?

Reply:
It's fine. Just the use of the word width makes me think of step size rather than the width of a filter/band.